### ISSUE TITLE
v0.10.0 miscategorizes uncalled function as unreachable code

### ISSUE THREAD

**Issue by matthewpersico** [koalaman/shellcheck#3138]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2329
- My shellcheck version (`shellcheck --version` or 'online'): [v0.10.0](https://github.com/koalaman/shellcheck/releases/tag/v0.10.0)
- [ ] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [x] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [ ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

main()
{
        echo I am in main
}

subber ()
{
        echo I am in subber
}

main
exit

```

#### Here's what shellcheck [v0.10.0](https://github.com/koalaman/shellcheck/releases/tag/v0.10.0) currently says:
In /home/mpersico5/sctest line 10:
        echo I am in subber
        ^-----------------^ SC2317 (info): Command appears to be unreachable. Check usage (or ignore if invoked indirectly).

#### Here's what I wanted or expected to see:
```
Line 8	SC2329: This function is never invoked. Check usage (or ignored if invoked indirectly).
```
Can we get a new release v0.11.0? I cannot bring unreleased/untagged HEAD code into my org. Thank you.

**Comment by matthewpersico** [comment]
From my local testing, it appears that v0.9.0 has the same issue.

**Comment by brother** [comment]
Dupe of #3137
